docs/explanation/testing.md

```diff
@@ -98,53 +151,78 @@ It builds a fixture called `ops_test` that helps you interact with Juju through
 
 It also provides convenient markers and command line parameters (e.g., the `@pytest.mark.skip_if_deployed` marker in combination with the `--no-deploy` configuration helps you skip, e.g., a deployment test in the case where you already have a deployment).
 
-
 > See more:
 > - [`pytest-operator` > Markers](https://github.com/charmed-kubernetes/pytest-operator/blob/main/docs/reference.md#markers)
 > - [`pytest-operator` > Command line parameters](https://github.com/charmed-kubernetes/pytest-operator/blob/main/docs/reference.md#command-line-parameters)
 
-
-
-**Examples.**
+### Examples
 
 - [https://github.com/canonical/prometheus-k8s-operator/blob/main/tests/integration/test_charm.py](https://github.com/canonical/prometheus-k8s-operator/blob/main/tests/integration/test_charm.py)
 
----
-
-(interface-tests)=
-## Interface tests
-> See also: {ref}`manage-interfaces`
-
-Interface tests are tests that verify the compliance of a charm with an interface specification.
-Interface specifications, stored in {ref}`charm-relation-interfaces <charm-relation-interfaces>`, are contract definitions that mandate how a charm should behave when integrated with another charm over a registered interface.
-
-Interface tests will allow `charmhub` to validate the relations of a charm and verify that your charm indeed supports "the" `ingress` interface and not just an interface called "ingress", which happens to be the same name as "the official `ingress` interface v2" as registered in charm-relation-interfaces (see [here](https://github.com/canonical/charm-relation-interfaces/tree/main/interfaces/ingress/v2)).
+## Continuous integration
 
-Also, they allow alternative implementations of an interface to validate themselves against the contractual specification stored in charm-relation-interfaces, and they help verify compliance with multiple versions of an interface.
+Typically, you want the tests to be run automatically against any PR into your repository's main branch, and potentially trigger a new release whenever the tests succeed. Continuous deployment is out of scope for this page, but we will look at how to set up basic continuous integration.
 
-An interface test is a contract test powered by [`ops.testing`](ops_testing) and a pytest plugin called [`pytest-interface-tester`](https://github.com/canonical/pytest-interface-tester). An interface test has the following pattern: 
-1) **GIVEN** an initial state of the relation over the interface under test
-2) **WHEN** a specific relation event fires
-3) **THEN** the state of the databags is valid (e.g. it satisfies an expected pydantic schema)
+Create a file called `.github/workflows/ci.yaml`. For example, to include a `lint` job that runs the `tox` `lint` environment:
 
-On top of databag state validity, one can check for more elaborate conditions.
+```yaml
+name: Tests
+on:
+  workflow_call:
```

Comment:
Should we make this more realistic, like including `pull_request` and `push: \ branches: \ -main`?